pkg/kademlia/kademlia.go

```diff
@@ -0,0 +1,36 @@
+package kademlia
+
+import (
+	"context"
+
+	"storj.io/storj/protos/overlay"
+)
+
+type nodeID string
+
+// DHT is the interface for the DHT in the Storj network
+type DHT interface {
+	GetNodes(ctx context.Context) []overlay.Node
+	GetRoutingTable(ctx context.Context) RoutingTable
+	Bootstrap(ctx context.Context) error
+	Ping(ctx context.Context, node overlay.Node) overlay.Node
+	FindNode(ctx context.Context, ID nodeID) overlay.Node
+	FindValue(ctx context.Context)
+}
+
+// RoutingTable contains information on nodes we have locally
+type RoutingTable interface {
+	GetBuckets() []*Bucket
+	GetBucket(id string) (ok bool, bucket Bucket)
```

Comment:
GetBuckets() could return lots of things. maybe make it take a callback per bucket?
returning ok bool first before bucket seems unusual

overall i'd probably make this a bit more high level - 

```
type RoutingTable interface {
  // local params
  LocalId() nodeId
  K() int
  CacheSize() int

  GetBucket(id string) (bucket Bucket, ok bool)
  GetBuckets() []Bucket

  FindNear(id nodeId, limit int) ([]overlay.Node, error)
  
  ConnectionSuccess(id string, address overlay.NodeAddress)
  ConnectionFailed(id string, address overlay.NodeAddress)

  // these are for refreshing
  SetBucketTimestamp(id string, now time.Time) error
  GetBucketTimestamp(id string, bucket Bucket) (time.Time, error)
}
```

Reply:
I'm not understanding 
```
GetBuckets(cb func(Bucket) (keep_going bool))
```

Why would get buckets just return an array of buckets ?

Reply:
sorry, you're totally right, there's a fixed number of DHT buckets. it's not actually capped at 256 (there's edge cases where a non-rightmost bucket might split in the paper), but it's pretty close, and that totally doesn't need pagination